dcpy/lifecycle/ingest/extract.py

```diff
@@ -19,62 +17,64 @@
 from . import TMP_DIR, configure
 
 
-def download_file_from_source(template: Template, version: str, dir=TMP_DIR) -> Path:
-    """From parsed config template and version, download raw data from source
-    and return path of saved local file."""
-    match template.source:
+def download_file_from_source(
+    source: Source, filename: str, version: str, dir=TMP_DIR
+) -> None:
+    """
+    From parsed config template and version, download raw data from source to provided path
+    """
+    path = dir / filename
+    match source:
         ## Non reqeust-based methods
         case LocalFileSource():
-            return template.source.path
+            pass
```

Comment:
Thought it felt a little silly to pass in the full path and also return it. Though actually, with a default arg for dir it's a bit odd. Need to think about that for a moment

Reply:
returning `None` seems nice. there's no default for `filename` and in either case of a provided or default `dir`, those two values would be known by whatever calls this function. so returning the path doesn't seem worth it